src/SpinViewer/SpinViewer.ts

```diff
@@ -0,0 +1,289 @@
+import Component from "@egjs/component";
+import Axes, { PanInput } from "@egjs/axes";
+import SpriteImage from "./SpriteImage";
+import { VERSION } from "../version";
+
+const DEFAULT_PAN_SCALE = 0.21;
+
+/**
+ * @memberof eg.view360
+ * @extends eg.Component
+ * SpinViewer
+ */
+class SpinViewer extends Component<{
+  /**
+   * Events that occur when component loading is complete
+   * @ko 컴포넌트 로딩이 완료되면 발생하는 이벤트
+   * @name eg.view360.SpinViewer#load
+   * @event
+   * @param {Object} param The object of data to be sent to an event <ko>이벤트에 전달되는 데이터 객체</ko>
+   * @param {HTMLElement} param.target The target element for which to display the image <ko>이미지를 보여줄 대상 엘리먼트</ko>
+   * @param {HTMLElement} param.bgElement Generated background image element <ko>생성된 background 이미지 엘리먼트</ko>
+   *
+   * @example
+   *
+   * viwer.on({
+   *  "load" : function(evt) {
+   *    this.spinBy(360, {duration: 300});
+   *  }
+   * });
+   */
+  load: {
```

Comment:
It would be better to separate the events by type.

Reply:
What's the benefit of doing that?

Reply:
You can get the type into the callback function.

```js
funtion callback(e: OnLoad) {
}
```

Reply:
Okay, but I prefer not separating them all as there're some event names shared by classes.

```ts
export type SpinViewerEvent = {
  load: {
    target: HTMLElement;
    bgElement: HTMLDivElement;
  }
}

function onLoad(e: SpinViewerEvent["load"]) {
}
```

Also, in that case default properties are gone from Component. This should be handled on @egjs/component.

![image](https://user-images.githubusercontent.com/26213435/98216091-f407d000-1f8b-11eb-8ddb-4a287e1db735.png)
![image](https://user-images.githubusercontent.com/26213435/98216108-fb2ede00-1f8b-11eb-9eee-84bed40306c3.png)

Reply:
Good.